Remove inferred conditional copyability section (#3)
kavon/swift-evolution: proposals/NNNN-noncopyable-generics.md

But SE-0390 also made a number of concessions to simplify the initial implementation,
limitations which significantly reduce the usefulness of noncopyable types.

> For example, SE-0390 did not allow noncopyable types to be used in generics,
> which prevents them from being used with `Optional`,
> which prevents you from defining a failable initializer on a noncopyable type:

Reviewer: ```suggestion
For example, SE-0390 did not allow noncopyable types to be used in generics,
which prevents them from being used with `Optional`,
which then prevents you from defining a failable initializer on a noncopyable type:
```

Author: Will be in next PR.

In addition, comments with _generic signature_ or _interface type_ may appear next to a type or function, respectively.
These generic signatures will detail all of the requirements or type constraints for generic parameters in scope:
```swift
// signature <Self where Self: Comparable & Copyable>
protocol P: Comparable {}

struct S<T: Equatable> {
  func f() {}  // signature <T where T: Equatable & Copyable>
}
```

#### Conforming to `Copyable`

> `Copyable` has no explicit requirements.
> Value types like structs and enums conform to `Copyable` when all of their stored properties or associated values conform to `Copyable`.

Reviewer: ```suggestion
`Copyable` has no explicit requirements.
Value types like structs and enums conform to `Copyable` when they do not contain a `deinit` and all of their stored properties or associated values conform to `Copyable`.
```

Author: Good catch!  I'll fold this into my next PR.

protocol CasinoChip: Token, ~Copyable {}
```

> Note that `~Copyable` as a constraint is not viral:
> types that conform to a `~Copyable` protocol can themselves be `Copyable`.
> Similarly, a protocol that inherits from a noncopyable one can still be `Copyable`.

Reviewer: ```suggestion
Note that `~Copyable` on a protocol is not viral:
types that conform to a `~Copyable` protocol can themselves be `Copyable`.
Similarly, a protocol that inherits from a noncopyable one can still require its
conformers to be `Copyable`.
```

> Note that `~Copyable` as a constraint is not viral:
> types that conform to a `~Copyable` protocol can themselves be `Copyable`.
> Similarly, a protocol that inherits from a noncopyable one can still be `Copyable`.

Reviewer: Clarifying that it's not protocols themselves that are Copyable, but may or may not _require_ Copyable. Existentials constrained by even a noncopyable protocol will also be Copyable unless you opt-out with `~Copyable`. Not sure if that's already covered or not.

```suggestion
Note that `~Copyable` as a constraint is not viral:
types that conform to a `~Copyable` protocol can themselves be `Copyable`.
Similarly, a protocol can still require its conformers to be `Copyable` even if it inherits
from a `~Copyable` protocol.
```

Reviewer: I think it is covered in the existentials section.

This section spells out additional details about the proposed extensions.

### Conditionally-copyable types

> An aggregate type can only be copyable if all its stored fields are copyable.

Reviewer: Sticking with "value" instead of "aggregate" type to contrast from classes. Adding the deinit rule as well.
```suggestion
A value type can only be copyable if all its stored fields are copyable
and it does not define a deinitializer.
```
